ruby/lib/google/protobuf.rb

```diff
@@ -37,6 +37,7 @@ module Google
   module Protobuf
     class Error < StandardError; end
     class ParseError < Error; end
+    class TypeError < ::TypeError; end
```

Comment:
shouldn't this inherit from Google::Protobuf::Error instead of ruby's TypeError?

Reply:
Possibly, but that would make this change a breaking one. Any code that is currently relying on this being a `TypeError` would start failing if the parent class is changed. By making this a subclass of the standard `TypeError` existing rescues should continue to work, but folks can opt in to catching the more specific class now.

Technically you could make TypeError the "base" class here (TypeError inherits from StandardError) instead of StandardError and it would then allow someone to rescue any Protobuf::Error and it should still be backwards compatible.

```Ruby
def foo; raise TypeError; rescue StandardError => e; puts "caught"; end
>> foo
caught
```

```Ruby
module Google
  module Protobuf
    class Error < ::TypeError; end
    class ParseError < Error; end
    class TypeError < Error; end
  end
end
```